src/java_bytecode/java_bytecode_parse_tree.h

```diff
@@ -173,8 +173,34 @@ class java_bytecode_parse_treet
     bool is_abstract=false;
     bool is_enum=false;
     bool is_public=false, is_protected=false, is_private=false;
+    bool read_attribute_bootstrapmethods = false;
     size_t enum_elements=0;
 
+    enum class method_handle_typet
+    {
+      BOOTSTRAP_METHOD_HANDLE,
+      BOOTSTRAP_METHOD_HANDLE_ALT,
+      LAMBDA_METHOD_HANDLE,
+      UNKNOWN_HANDLE
+    };
+
+    struct lambda_method_handlet
```

Comment:
isn't the main difference that struct members are public by default while class members are private ? But yes, no problem to make this a class.

Reply:
Yes, C++ doesn't make a large differentiation between the struct and class keywords. I am asking for a class to be used with each of the fields accessed through getter and setter methods instead of publicly accessible fields. The production of the debug text on lines 1480-1485, could then be defined as a method instead of inline at the point of building the class.
I know this is a relatively small example of a class to build with getters/setters which feel like boiler plate code. However I think it gives a superior foundation to build on, leading to more compartmentalised code in the end result, instead of large individual files.